app/models/patient.rb

```diff
@@ -931,4 +935,132 @@ def new_submission_token
     end
     token
   end
+
+  def handle_update
+    monitoring_change if monitoring_changed?
+    isolation_change if isolation_changed?
+    case_status_change if case_status_changed?
+    symptom_onset_change if symptom_onset_changed?
+    continuous_exposure_change if continuous_exposure_changed?
+  end
+
+  def monitoring_change
+    return if monitoring
+
+    self.closed_at = DateTime.now
+    self.continuous_exposure = false
+  end
+
+  def isolation_change
+    return if isolation
+
+    self.extended_isolation = nil
+    # NOTE: The below will overwrite any new value they may set for symptom onset as they can not be set in the exposure workflow.
+    self.user_defined_symptom_onset = false
+    self.symptom_onset = calculated_symptom_onset
+  end
+
+  def case_status_change
+    return unless ['Suspect', 'Unknown', 'Not a Case'].include?(case_status) && public_health_action != 'None'
+
+    self.public_health_action = 'None'
+  end
+
+  def symptom_onset_change
+    return unless symptom_onset.nil?
+
+    self.user_defined_symptom_onset = false
+    self.symptom_onset = calculated_symptom_onset
+  end
+
+  def continuous_exposure_change
+    self.continuous_exposure = false if continuous_exposure && !monitoring
+  end
+
+  def monitoring_history_edit(history_data, diff_state)
```

Comment:
I'd ask that this method (and those above) also have method comments. For the ones above a comment describing the cases and why the additional changes are made would be ideal.